### PR TITLE
Pr 447

### PR DESCRIPTION
**Pull request by ivan-015**
This is a copy of PR #447, I made a personal branch to solve the conflicts in that branch. I will close #447 to merge this PR in.

### CODE REVIEW

File: lib/generation/generators/symbols/pb_instancesym_gen.dart [hunk 1]
```diff
@@ -33,8 +32,8 @@ class PBSymbolInstanceGenerator extends PBGenerator {
       buffer.write('  builder: (context, constraints) {\n');
       buffer.write('    return ');
 
-      // If storage found an instance, get its master
-      var masterSymbol =
+      // If we are processing master symbol generation grab the master, else grab gab the master of ourselves
```
Comment (SushiRoll53) [inline lib/generation/generators/symbols/pb_instancesym_gen.dart]: small `gab` typo

File: lib/interpret_and_optimize/services/pb_shared_aggregation_service.dart [hunk 2]
```diff
@@ -47,22 +47,33 @@ class PBSharedInterAggregationService {
...
-      if ((prop.value != null) && (prop.type == PBSharedInstanceIntermediateNode)) {
-        ///if the [PBSharedMasterNode] contains [PBSharedInstanceIntermediateNode] as parameters
-        ///then its going gather the information of its [PBSharedMasterNode].
-        gatherSharedValues(prop.value);
-      }
     }
-    sharedMasterNode.overridableProperties
-        .removeWhere((prop) => prop == null || prop.value == null);
+
+    //sharedMasterNode.overridableProperties
```
Comment (SushiRoll53) [inline lib/interpret_and_optimize/services/pb_shared_aggregation_service.dart]: If it is commented out, we probably should remove it

File: lib/interpret_and_optimize/value_objects/pb_symbol_master_params.dart [hunk 3]
```diff
@@ -30,7 +30,9 @@ class PBSymbolMasterParameter extends PBVisualIntermediateNode
       this.bottomRightX,
       this.bottomRightY,
       {this.context})
-      : super(Point(0, 0), Point(0, 0), context, name);
+      : super(Point(0, 0), Point(0, 0), context, name) {
+
```
Comment (SushiRoll53) [inline lib/interpret_and_optimize/value_objects/pb_symbol_master_params.dart]: if the constructor is empty, we should just use `;` instead

**Review by SushiRoll53 — APPROVED** [review]
LGTM, just small things to fix

File: lib/generation/generators/middleware/state_management/utils/middleware_utils.dart [hunk 4]
```diff
@@ -77,16 +77,18 @@ class MiddlewareUtils {
...
-
-      // default provider event handler for gestures.
-      void onGesture() {
-      }
+        LayoutBuilder currentLayout;
+        final String widgetName;
+        ${defaultStateName}(this.widgetName);
+  
+        // default provider event handler for gestures.
+        void onGesture() {
```
Comment (siliconivan) [inline lib/generation/generators/middleware/state_management/utils/middleware_utils.dart:85]: What is this empty method for?

**Review by siliconivan — APPROVED** [review]
lgtm

**Pull request merged**